### ISSUE TITLE
Incorrect GFLOPs calculation in model_info (reports ~16.4 GFLOPs instead of official 15.8 GFLOPs)

### ISSUE THREAD

**Issue by 733ly** [ultralytics/yolov5#13590]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

When I ran the model previously, all of the printouts showed the YOLOv5s model as having 7,015,519 parameters and 15.8 GFLOPs. Today, after fusing layers, it suddenly reports 16.4 GFLOPs, and even after freshly cloning the repository it still shows 16.4. Why is this happening?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @733ly, thank you for bringing this up and for your detailed report! ⚡️ This is an automated response to help you get started—an Ultralytics engineer will also review and assist you soon.

If this is a 🐛 Bug Report, please provide a **minimum reproducible example** (MRE) that demonstrates the GFLOPs calculation issue. This helps us debug efficiently.

For general questions or custom training, please include as much supporting information as possible, such as environment details, command outputs, and relevant logs. Also, check our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) for helpful guidance.

## Requirements
[**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed, including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To set up YOLOv5:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Environments
YOLOv5 can be run in any of the following up-to-date, verified environments (with all dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>
If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py), and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your patience while we look into this! 😊

**Comment by pderrenger** [comment]
Hi @733ly,

Thank you for reporting this discrepancy. You're correct that there's a difference between the reported GFLOPs (16.4) and the previously documented value (15.8). This can happen for several reasons:

1. The `model_info()` function calculates FLOPs dynamically based on the current model architecture
2. Layer fusion can affect FLOPs calculations as the model structure changes
3. Recent YOLOv5 updates may have slightly modified the architecture or calculation method

Different FLOPs calculation tools and methods often produce variations in the reported values. The calculation depends on how operations are counted and which operations are included in the total.

If you need a consistent metric for benchmarking or documentation purposes, you might want to specify which calculation tool and method you're using, as the exact calculation can vary between implementations.

Paula Derrenger

**Comment by 733ly** [comment]
Hi@pderrenger，
 Thank you for answering my question. I’m still a bit confused, though. First, I didn’t modify the original yolov5s.yaml, yet the reported GFLOPs jumped to 16.4. Second, I had my junior colleague verify this on his machine, and he also got 16.4. So why would the locally downloaded YOLOv5 code change simultaneously on different machines—could it be that the code is managed remotely rather than truly local? Finally, does this shift mean I need to redo all my experiments? After all, if I’m pursuing a lightweight improvement, metrics like parameter count and GFLOPs should still serve as meaningful references. Thanks again for your help, and I wish you all the best!

**Comment by pderrenger** [comment]
Hi @733ly,

To answer your questions:

1. The GFLOPs calculation likely changed due to a recent GitHub repository update. When you clone or pull from GitHub, you get the latest code with any recent modifications to the calculation methods.

2. Since you and your colleague both pulled the latest code version, you're seeing the same updated GFLOPs calculation on different machines.

3. For lightweight model comparisons, I suggest documenting which YOLOv5 version you used for each experiment (git commit hash). What's most important is consistency within your comparative experiments - either compare all models using the old calculation method or all using the new one.

If you need to maintain exact consistency with previous work, you could revert to a specific earlier commit of the repository where the GFLOPs calculation was 15.8.

Paula Derrenger

**Comment by 733ly** [comment]
Hi @pderrenger ，
Thank you again for answering my question. I should clarify that neither my colleague nor I downloaded a newer version—our yolov5-master was cloned several months ago. When I noticed the issue two days ago and asked him to check, he found that running yolo.py with the same yolov5s.yaml now also reports 16.4 GFLOPs instead of the 15.8 he had originally recorded. The only proof we have of our original model is the trained **.pt** file, whose parameters and GFLOPs still match our earlier records.Thanks again for your help, and I wish you all the best!

**Comment by pderrenger** [comment]
Hi @733ly,

That's quite puzzling since Git repositories don't update automatically without explicit commands. Here are some possible explanations for the GFLOPs discrepancy:

1. Environment differences - Python package updates (like torch or numpy) might affect calculations
2. Input resolution changes - Even slight differences in input size affect GFLOPs calculations
3. Inadvertent code modifications - Check for any local changes with `git status`
4. Different model mode - Inference vs training mode can affect reported metrics

You can verify the exact version using `git log -1` and compare hash values between machines. For your research, I recommend using your trained .pt file as the reference point since it maintains consistent metrics, and documenting all calculation parameters in your methodology.

Paula Derrenger
